productcat/src/main/java/com/visenze/productcat/android/data/DataCollection.java

```diff
@@ -39,7 +39,11 @@ private String getIMEI(Context context) {
         TelephonyManager tm = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);
         if (null != tm) {
             if (ActivityCompat.checkSelfPermission(context, Manifest.permission.READ_PHONE_STATE) == PackageManager.PERMISSION_GRANTED) {
-                deviceUniqueIdentifier = tm.getDeviceId();
+                try {
+                    deviceUniqueIdentifier = tm.getDeviceId();
+                } catch (Exception e) {
+                    return deviceUniqueIdentifier;
```

Comment:
can print out the exception in the log, otherwise no idea what happens

Reply:
added already